From a5b14bdddfafc7a7fbdc98d6635c16bb3a359b95 Mon Sep 17 00:00:00 2001
From: Cyrille Fleury <cyrille.fleury@nxp.com>
Date: Thu, 18 Apr 2019 13:19:21 +0200
Subject: [PATCH 3/9] MMIOT-244:On secure closed board, linux can't configure
 SNVS register NPSWA_EN. It must be done in ATF

Signed-off-by: Cyrille Fleury <cyrille.fleury@nxp.com>
---
 plat/imx/imx8mq/imx8mq_bl31_setup.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plat/imx/imx8mq/imx8mq_bl31_setup.c b/plat/imx/imx8mq/imx8mq_bl31_setup.c
index 49f08cd8f..dd4b4277b 100644
--- a/plat/imx/imx8mq/imx8mq_bl31_setup.c
+++ b/plat/imx/imx8mq/imx8mq_bl31_setup.c
@@ -173,6 +173,8 @@ static uint32_t get_spsr_for_bl33_entry(void)
 
 #define GPR_TZASC_EN		(1 << 0)
 #define GPR_TZASC_EN_LOCK	(1 << 16)
+#define SNVS_HPCOMR             0x04
+#define SNVS_NPSWA_EN           (1 << 31)
 unsigned int freq;
 
 void system_counter_init(void)
@@ -192,6 +194,9 @@ void bl31_tzc380_setup(void)
 {
 	unsigned int val;
 
+	val = mmio_read_32(IMX_SNVS_BASE + SNVS_HPCOMR);
+	mmio_write_32(IMX_SNVS_BASE + SNVS_HPCOMR, val | SNVS_NPSWA_EN);
+
 	val = mmio_read_32(IMX_IOMUX_GPR_BASE + 0x28);
 	if ((val & GPR_TZASC_EN) != GPR_TZASC_EN)
 		return;
-- 
2.17.1

